src/fragmentarium/ui/info/genres.ts

```diff
@@ -0,0 +1,117 @@
+import _ from 'lodash'
+
+export interface Tree {
+  value: string
+  nodes?: ReadonlyArray<Tree>
+}
+export function flattenTree(genres: Tree) {
```

Comment:
This should be refactored as codeclimate says. Visitor pattern might help.

I fail to see why "flattening" would require a search. It should be enough to traverse the structure in order.